### PR TITLE
[TXP-27] Add feedback buttons to experiment cards

### PR DESCRIPTION
**Pull request by lmorchard**
Think I have a good first stab at adding the feedback button, including metrics and l10n and all that fun stuff. Some remaining questions:

- [x] Needs a hover state? (/cc @johngruen)

Also added a React Storybook install here. Not perfect, but you should be able to see what it's about like so:

* Start a dev server like usual with `npm start`
* In a second shell, start Storybook with `npm run storybook`
* Visit http://localhost:6006/ to hopefully see this:

capture

### CODE REVIEW

**Comment by lmorchard** [comment]
Remaining question I have: Does this feedback button need a hover state? I didn't see one in #2462

**Comment by lmorchard** [comment]
Huh, looks like the test failure is in one of the integration tests. But, I'm not sure why it's failing.

**Comment by johngruen** [comment]
@lmorchard probably you can use the same hover as the `primary` class in the _buttons.scss file. The `button` class in that file also describes transition timing for backgrounds

**Comment by lmorchard** [comment]
> Huh, looks like the test failure is in one of the integration tests. But, I'm not sure why it's failing.

[Ah hah!](https://circleci.com/gh/lmorchard/testpilot/833) It was because [I changed markup](https://github.com/mozilla/testpilot/pull/2548/files#diff-57f336f2a2f666d546ecb9273f19fafeR61) and [the CSS selector for experiment name in the home page class](https://github.com/mozilla/testpilot/blob/e3edbabb6d9673b03cb664e8025686ea6ee39b70/frontend/test/ui/pages/desktop/home.py#L47) no longer matched. Really weird that the error caused by that would be [a unicode exception](https://circleci.com/gh/lmorchard/testpilot/831), though!

**Comment by lmorchard** [comment]
Just to make this more easily merged, I rebased out the storybook stuff since it has its own PR in #2549

File: frontend/src/app/components/ExperimentRowCard.js [hunk 1]
```diff
@@ -79,6 +86,32 @@ export default class ExperimentRowCard extends React.Component {
     );
   }
 
+  renderFeedbackButton() {
+    if (!this.props.enabled) { return ''; }
```
Comment (fzzzy) [inline frontend/src/app/components/ExperimentRowCard.js]: I usually use null to indicate nothing should be rendered, but it's not that important.

File: frontend/tasks/content.js [hunk 2]
```diff
@@ -161,7 +161,7 @@ function buildExperimentsData() {
 }
 
 .experiment-icon-${experiment.slug} {
-  background-image: url(${experiment.thumbnail});
+  background-image: url(../..${experiment.thumbnail});
```
Comment (fzzzy) [inline frontend/tasks/content.js]: Why is this changing?

File: locales/en-US/app.ftl [hunk 3]
```diff
@@ -94,6 +94,7 @@ experimentListHeader = Pick your experiments!
 
 // An individual experiment in the listing of all Test Pilot experiments.
 [[experimentCard]]
+experimentCardFeedback = Feedback
```
Comment (fzzzy) [inline locales/en-US/app.ftl]: I think we need to let flod know that there are going to be new strings.

**Review by fzzzy — APPROVED** [review]
Looks good.

File: frontend/tasks/content.js (hunk 2 above)
Comment (lmorchard) [reply frontend/tasks/content.js]: Kind of a hack, [but a relative path works much better with the Storybook deployment script](http://storybook.testpilot.dev.mozaws.net/6a7fca5b3e30612dd561adf631ae28641450ef8e/?knob-Has%20Addon=false&knob-Enabled=false&knob-Experiment=%7B%22title%22%3A%22Sample%20experiment%22%2C%22description%22%3A%22This%20is%20an%20example%20experiment%22%2C%22subtitle%22%3A%22%22%2C%22slug%22%3A%22snooze-tabs%22%2C%22survey_url%22%3A%22https%3A%2F%2Fexample.com%22%2C%22installation_count%22%3A%2286753%22%2C%22created%22%3A%222010-06-21T12%3A12%3A12Z%22%2C%22modified%22%3A%222010-06-21T12%3A12%3A12Z%22%7D&selectedKind=ExperimentRowCard&selectedStory=has%20addon%20%26%20enabled&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel).

File: locales/en-US/app.ftl (hunk 3 above)
Comment (lmorchard) [reply locales/en-US/app.ftl]: @flodolo New string here :) Also kind of wondering if we can tie Pontoon into [the Storybook view](http://storybook.testpilot.dev.mozaws.net/6a7fca5b3e30612dd561adf631ae28641450ef8e/?knob-Has%20Addon=false&knob-Enabled=false&knob-Experiment=%7B%22title%22%3A%22Sample%20experiment%22%2C%22description%22%3A%22This%20is%20an%20example%20experiment%22%2C%22subtitle%22%3A%22%22%2C%22slug%22%3A%22snooze-tabs%22%2C%22survey_url%22%3A%22https%3A%2F%2Fexample.com%22%2C%22installation_count%22%3A%2286753%22%2C%22created%22%3A%222010-06-21T12%3A12%3A12Z%22%2C%22modified%22%3A%222010-06-21T12%3A12%3A12Z%22%7D&selectedKind=ExperimentRowCard&selectedStory=has%20addon%20%26%20enabled&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel) someday to make in-place localization of specific site states easier

Comment (flodolo) [reply locales/en-US/app.ftl]: Can you add some context (and a comment) on what this string is? Is it a button label (e.g. "Give feedback?"), a link, something else?

For the latter, you might want to talk with Matjaz ;-)

**Comment by lmorchard** [comment]
As a further trial, I rebased another tweak to the Storybook deployment to get this link working:

http://storybook.testpilot.dev.mozaws.net/pull/2548/

File: frontend/src/app/components/ExperimentRowCard.js (hunk 1 above)
Comment (lmorchard) [reply frontend/src/app/components/ExperimentRowCard.js]: Hmm, I thought I saw an error from React when I used null, but that might have been something else. I'll change it to be consistent

**Comment by lmorchard** [comment]
Just noticed this has an r+ - going to snip out the storybook deployment thing since it's in [another PR](https://github.com/mozilla/testpilot/pull/2567). Then, probably merge this

**Comment by SoftVision-PaulOiegas** [comment]
I don't see any feedback buttons on dev. 🤔

**Comment by lmorchard** [comment]
Do you have experiments enabled on dev? The feedback button only appears on enabled experiments

**Comment by SoftVision-PaulOiegas** [comment]
I see them now. I thought this will be displayed even if experiments are not enabled. Thanks @lmorchard for clarifying.

**Comment by ChenMorpheus** [comment]
@lmorchard 
Thanks for implementing the feedback button. However, I was wondering if we can tweak it a little bit. There are several suggestions for the UI:
1. Move down the icon and "feedback" text 1 px to make it visually align with the button vertically. 
2. Have 8 px gap on the right end of the feedback button.
3. Make the feedback button align vertically with the experiment title.
You can find the measurement below. Please let me know if there is anything unclear. Thanks.

**Comment by lmorchard** [comment]
This is a merged PR, can you open a new issue for these changes? (I may or may not be the one who picks up the task, and we'll want to triage it & etc)

**Comment by ChenMorpheus** [comment]
@lmorchard Thanks for the suggestion. I've opened a new issue at #2629 , let me know if any questions, thanks.

**Pull request merged**